### ISSUE TITLE
[a11y]: Overflowmenu component's IconElement fails Accessibility Checker

### ISSUE THREAD

**Issue by kaushalnavneet** [carbon-design-system/carbon#13820]
### Package

@carbon/react

### Package version

v10.58.6

### React version

v7.59.6

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

I wasn't expecting this AVT error to appear: `The ARIA attributes "aria-label" are not valid for the element <span> with implicit ARIA role "generic"`
When looked into a bit, it is coming right from [this aria-label](https://github.com/carbon-design-system/carbon/blob/0571ef37b882a885967d41d47896b280d5bca29e/packages/react/src/components/OverflowMenu/OverflowMenu.js#L562) of `Overflowmenu` component.

Issue: issue 5 of [this AVT issue](https://github.ibm.com/org-ids/roadmap/issues/24936#issue-33678073) against Pipeline

![ce50d983-4599-4cab-89ae-297e8e32536e](https://github.com/carbon-design-system/carbon/assets/44249850/983ca7f0-d0de-4666-9e1c-911fc1e0f3cd)

### Reproduction/example

explained in the description

### Steps to reproduce

This is somewhat we are using in our code:

```
<OverflowMenu
      ariaLabel=""
      className="bx--btn ocr--property-add-btn pull-right"
      flipped
      iconDescription=""
      id="add-property"
      renderIcon={iconProps => (
        <span {...iconProps} className="bx--btn bx--btn--primary" tabIndex="0">
          "Add"
          <ChevronDown16 className="bx--btn__icon" />
        </span>
      )}
      selectorPrimaryFocus={selectorPrimaryFocus}
      tabIndex="-1"
      title="some title"
    >
      {renderOverflowItems() <some function to return the menu items>}
    </OverflowMenu>

```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hey @kaushalnavneet, `iconProps` is meant to be placed on the icon, not on a span containing text. The [OverflowMenu design specification](https://carbondesignsystem.com/components/overflow-menu/style) does not include text as part of the supported spec, so `renderIcon` isn't intended to be used for this purpose. I think the [experimental MenuButton](https://react.carbondesignsystem.com/?path=/story/experimental-unstable-menubutton--default) might be a better fit for this use case.

That aside, you might try moving `...iconProps` to be on the icon instead of the `span`. Or you could use [rest syntax in object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Rest_in_Object_Destructuring) to remove the `aria-label` from `iconProps`.
